vernac/vernacentries.ml

```diff
@@ -1523,26 +1523,34 @@ let vernac_set_option ~locality table v = match v with
       vernac_set_option0 ~locality table v
 | _ -> vernac_set_option0 ~locality table v
 
-let vernac_add_option key lv =
-  let f = function
-    | StringRefValue s -> (get_string_table key).add (Global.env()) s
-    | QualidRefValue locqid -> (get_ref_table key).add (Global.env()) locqid
+(* We have to request twice the same polymorphic function as argument
+   because OCaml won't compute the same type for it in both cases. *)
```

Comment:
We could alternatively use a record, I'll push an example commit

Also maybe iter_table should live in goptions?

Reply:
Yes, I didn't like the repetition, your commit makes things much nicer.

Moving `iter_table` to `goptions` sounds relevant too, even if for now `vernacentries` would be the only user. I'll take care of doing the move.

Reply:
Done. Let me know what you think of this last commit.